atc/job_config_test.go

```diff
@@ -502,6 +502,86 @@ var _ = Describe("JobConfig", func() {
 				})
 			})
 
+			Context("when a simple parallel plan is the first step", func() {
```

Comment:
I think this test, and the ones following, are mostly testing `get` in relation to `PlanSequence` and not `aggregate` or `parallel` specifically. But if the desire is to move from `aggregate` to `parallel` in the future, then I think it makes sense to just keep them?

Reply:
Same with the `exec_engine` tests. If the plan is to eventually move from `aggregate` to `parallel` then I think it makes sense to keep the tests; unless they can be cut completely?

Reply:
I think these tests are just covering the recursing through all the different types of steps which contain other steps. It probably makes sense to keep it, since we need to recurse through the `Parallel` fields as well. This is necessary for finding all the 'job inputs', which are determined by looking through the entire plan for `get` steps.